src/lib/utils/reverseHighlightedParts.ts

```diff
@@ -0,0 +1,39 @@
+import { HighlightedParts } from '../../types';
+import getHighlightFromSiblings from './getHighlightFromSiblings';
+
+const reverseHighlightedParts = (parts: HighlightedParts[]) => {
+  if (!parts.some(part => part.isHighlighted)) {
+    return parts.map(part => ({ ...part, isHighlighted: false }));
+  }
+
+  // #1 // Actual behavior, same as Autocomplete
+  //   return parts.map(part => ({ ...part, isHighlighted: !part.isHighlighted }));
+
+  // #2 // Sibiling matching behavior, match the highlight of the siblings
+  return parts.map((part, i) => ({
+    ...part,
+    isHighlighted: !getHighlightFromSiblings(parts, i),
+  }));
```

Comment:
@Haroenv 

I've split the logic in `getHighlightFromSiblings` so it returns either the matching siblings `isHighlighted` value (if they match) or the value of the current part.

This way we can use it in `reverseHighlightedParts` or `highlight`